app/lib/account/consent_backend.dart

```diff
@@ -219,9 +220,16 @@ class ConsentBackend {
 
   /// Returns the [Consent] for [consentId] and checks if it is for [user].
   Future<Consent> _lookupAndCheck(String consentId, User user) async {
-    final c = await _db.lookupValue<Consent>(
-        _db.emptyKey.append(Consent, id: consentId),
-        orElse: () => null);
+    final key = _db.emptyKey.append(Consent, id: consentId);
+    final c = await withRetryTransaction(_db, (tx) async {
+      final c = await tx.lookupValue<Consent>(key, orElse: () => null);
+      if (c == null) return null;
```

Comment:
Just throw `NotFoundException` as you would have done below... I'm pretty sure `withRetryTransaction` won't retry a `ResponseException` (but check the code).

IMO, we want `withRetryTransaction` to only retry datastore related exceptions, but we haven't exactly nailed down what these are yet.